interfaces/vault_kv/v0/README.md

```diff
@@ -0,0 +1,62 @@
+# `vault-kv`
+
+## Usage
+
+Some charms require a secure key value store. This relation interface describes the expected behavior of any charm claiming to interact with Vault Key Value stores.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- mount_suffix, egress_subnet --> Provider
+    Provider -- vault_url, mount, credentials --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+Provider expectations
+
+- Must provide the vault url
+- Must provide a key value mount, the mount name shall respect the following pattern: charm-<requirer app>-<user provided suffix>
+- Must provide a role-id and role-secret-id for each unit.
+  The role-id and role-secret-id shall be provided in the form of a Juju secret.
+  Usage of role-id and role-secret-id to access the KV backend shall be restricted to the unit's egress_subnet.
```

Comment:
In k8s we must assume that pods can get rescheduled, is this behavior gonna hold still?

Reply:
It's possible to modify the egress_subnet on the VaultKvRequires instanciation (what's implemented in the draft PR). Which send a vault-kv-relationship changed when the pod gets rescheduled. I'm not sure this is best practice within ops framework, but it does work as expected.

Reply:
It is possible not to have to respect this behaviour, however, we must relax the security constraint on credentials. (which would remove all the part about unit_egress from the interface)